src/main/java/com/palantir/docker/compose/connection/DockerMachine.java

```diff
@@ -100,9 +101,11 @@ public LocalBuilder withEnvironment(Map<String, String> newEnvironment) {
         public DockerMachine build() {
             dockerType.validateEnvironmentVariables(systemEnvironment);
             AdditionalEnvironmentValidator.validate(additionalEnvironment);
+            Map<String, String> combinedEnvironment = Maps.newHashMap();
+            combinedEnvironment.putAll(systemEnvironment);
+            combinedEnvironment.putAll(additionalEnvironment);
             Map<String, String> environment = ImmutableMap.<String, String>builder()
-                    .putAll(systemEnvironment)
-                    .putAll(additionalEnvironment)
+                    .putAll(combinedEnvironment)
```

Comment:
The two maps seem redundant/verbose. I propose you either return the `combinedEnvironment` or do `ImmutableMap.copyOf(combinedEnvironment)`.